File: app/src/main/java/io/github/webviewtemplate/MainActivity.java
package com.example.webviewapp;

import android.os.Bundle;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.webkit.WebViewClient;
import androidx.appcompat.app.AppCompatActivity;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class MainActivity extends AppCompatActivity {
    private WebView webView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        webView = new WebView(this);
        setContentView(webView);
        
        WebSettings s = webView.getSettings();
        s.setJavaScriptEnabled(true);
        s.setDomStorageEnabled(true);
        s.setAllowFileAccess(true);
        
        webView.setWebViewClient(new WebViewClient());
        
        String url = "https://yiresh.ia-box.com";
        try {
            BufferedReader r = new BufferedReader(new InputStreamReader(getAssets().open("url.txt")));
            String line = r.readLine();
            if(line != null && !line.trim().isEmpty()) url = line.trim();
            r.close();
        } catch (Exception e) {}
        
        webView.loadUrl(url);
    }
    @Override
    public void onBackPressed() {
        if(webView.canGoBack()) webView.goBack();
        else super.onBackPressed();
    }
}

File: app/src/main/java/io/github/webviewtemplate/MainActivity.kt
package com.example.webviewapp

import android.os.Bundle
import android.webkit.*
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {
    lateinit var webView: WebView
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        webView = WebView(this)
        setContentView(webView)
        webView.settings.javaScriptEnabled = true
        webView.settings.domStorageEnabled = true
        webView.webViewClient = object : WebViewClient() {
            override fun onPageFinished(view: WebView?, url: String?) {
                val filterJs = """
                (function(){
                  document.querySelectorAll('img').forEach(img=>{
                    if(img.width<100) return;
                    // כאן המסנן לפי הגדרתך - מרפק/ברך/צוואר
                    // גרסה 1 - מטשטש עד לבדיקה
                    img.style.filter = 'blur(15px)';
                  });
                })();
                """.trimIndent()
                view?.evaluateJavascript(filterJs, null)
            }
        }
        webView.loadUrl("https://www.google.com")
    }
}
